[2024-09-15 20:04:42,769] {taskinstance.py:1159} INFO - Dependencies all met for <TaskInstance: BRONZER_GET_STONCKS.BRONZE_TSLA scheduled__2024-01-24T00:00:00+00:00 [queued]>
[2024-09-15 20:04:42,773] {taskinstance.py:1159} INFO - Dependencies all met for <TaskInstance: BRONZER_GET_STONCKS.BRONZE_TSLA scheduled__2024-01-24T00:00:00+00:00 [queued]>
[2024-09-15 20:04:42,774] {taskinstance.py:1356} INFO - 
--------------------------------------------------------------------------------
[2024-09-15 20:04:42,774] {taskinstance.py:1357} INFO - Starting attempt 1 of 1
[2024-09-15 20:04:42,774] {taskinstance.py:1358} INFO - 
--------------------------------------------------------------------------------
[2024-09-15 20:04:42,784] {taskinstance.py:1377} INFO - Executing <Task(_PythonDecoratedOperator): BRONZE_TSLA> on 2024-01-24 00:00:00+00:00
[2024-09-15 20:04:42,786] {standard_task_runner.py:52} INFO - Started process 11734 to run task
[2024-09-15 20:04:42,788] {standard_task_runner.py:79} INFO - Running: ['airflow', 'tasks', 'run', 'BRONZER_GET_STONCKS', 'BRONZE_TSLA', 'scheduled__2024-01-24T00:00:00+00:00', '--job-id', '76', '--raw', '--subdir', 'DAGS_FOLDER/get_stocks.py', '--cfg-path', '/tmp/tmp5imgi0yr', '--error-file', '/tmp/tmpj0p8oif3']
[2024-09-15 20:04:42,789] {standard_task_runner.py:80} INFO - Job 76: Subtask BRONZE_TSLA
[2024-09-15 20:04:42,816] {task_command.py:370} INFO - Running <TaskInstance: BRONZER_GET_STONCKS.BRONZE_TSLA scheduled__2024-01-24T00:00:00+00:00 [running]> on host kaiwang.
[2024-09-15 20:04:42,855] {taskinstance.py:1569} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=BRONZER_GET_STONCKS
AIRFLOW_CTX_TASK_ID=BRONZE_TSLA
AIRFLOW_CTX_EXECUTION_DATE=2024-01-24T00:00:00+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2024-01-24T00:00:00+00:00
[2024-09-15 20:04:42,855] {taskinstance.py:1889} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/usr/lib/python3.10/pathlib.py", line 1175, in mkdir
    self._accessor.mkdir(self, mode)
FileNotFoundError: [Errno 2] No such file or directory: '/datalake/TSLA'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kaiwang/Documents/airflow_celeby/venv/lib/python3.10/site-packages/airflow/decorators/base.py", line 179, in execute
    return_value = super().execute(context)
  File "/home/kaiwang/Documents/airflow_celeby/venv/lib/python3.10/site-packages/airflow/operators/python.py", line 171, in execute
    return_value = self.execute_callable()
  File "/home/kaiwang/Documents/airflow_celeby/venv/lib/python3.10/site-packages/airflow/operators/python.py", line 189, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/home/kaiwang/Documents/airflow_celeby/dags/get_stocks.py", line 21, in get_history
    (Path(file_path).parent).mkdir(parents=True, exist_ok=True)
  File "/usr/lib/python3.10/pathlib.py", line 1179, in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
  File "/usr/lib/python3.10/pathlib.py", line 1175, in mkdir
    self._accessor.mkdir(self, mode)
PermissionError: [Errno 13] Permission denied: '/datalake'
[2024-09-15 20:04:42,858] {taskinstance.py:1395} INFO - Marking task as FAILED. dag_id=BRONZER_GET_STONCKS, task_id=BRONZE_TSLA, execution_date=20240124T000000, start_date=20240915T230442, end_date=20240915T230442
[2024-09-15 20:04:42,867] {standard_task_runner.py:92} ERROR - Failed to execute job 76 for task BRONZE_TSLA ([Errno 13] Permission denied: '/datalake'; 11734)
[2024-09-15 20:04:42,881] {local_task_job.py:156} INFO - Task exited with return code 1
[2024-09-15 20:04:42,891] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-09-15 20:04:42,895] {dagrun.py:547} ERROR - Marking run <DagRun BRONZER_GET_STONCKS @ 2024-01-24 00:00:00+00:00: scheduled__2024-01-24T00:00:00+00:00, externally triggered: False> failed
[2024-09-15 20:04:42,896] {dagrun.py:607} INFO - DagRun Finished: dag_id=BRONZER_GET_STONCKS, execution_date=2024-01-24 00:00:00+00:00, run_id=scheduled__2024-01-24T00:00:00+00:00, run_start_date=2024-09-15 23:04:37.442633+00:00, run_end_date=2024-09-15 23:04:42.896077+00:00, run_duration=5.453444, state=failed, external_trigger=False, run_type=scheduled, data_interval_start=2024-01-24 00:00:00+00:00, data_interval_end=2024-01-25 00:00:00+00:00, dag_hash=d881e529ddca44bbc90b852447cd3348
[2024-09-15 23:28:31,888] {taskinstance.py:1159} INFO - Dependencies all met for <TaskInstance: BRONZER_GET_STONCKS.BRONZE_TSLA scheduled__2024-01-24T00:00:00+00:00 [queued]>
[2024-09-15 23:28:31,893] {taskinstance.py:1159} INFO - Dependencies all met for <TaskInstance: BRONZER_GET_STONCKS.BRONZE_TSLA scheduled__2024-01-24T00:00:00+00:00 [queued]>
[2024-09-15 23:28:31,893] {taskinstance.py:1356} INFO - 
--------------------------------------------------------------------------------
[2024-09-15 23:28:31,893] {taskinstance.py:1357} INFO - Starting attempt 1 of 1
[2024-09-15 23:28:31,893] {taskinstance.py:1358} INFO - 
--------------------------------------------------------------------------------
[2024-09-15 23:28:31,903] {taskinstance.py:1377} INFO - Executing <Task(_PythonDecoratedOperator): BRONZE_TSLA> on 2024-01-24 00:00:00+00:00
[2024-09-15 23:28:31,906] {standard_task_runner.py:52} INFO - Started process 14413 to run task
[2024-09-15 23:28:31,908] {standard_task_runner.py:79} INFO - Running: ['airflow', 'tasks', 'run', 'BRONZER_GET_STONCKS', 'BRONZE_TSLA', 'scheduled__2024-01-24T00:00:00+00:00', '--job-id', '127', '--raw', '--subdir', 'DAGS_FOLDER/get_stocks.py', '--cfg-path', '/tmp/tmp6ghoqtvk', '--error-file', '/tmp/tmpj22xbktn']
[2024-09-15 23:28:31,908] {standard_task_runner.py:80} INFO - Job 127: Subtask BRONZE_TSLA
[2024-09-15 23:28:31,937] {task_command.py:370} INFO - Running <TaskInstance: BRONZER_GET_STONCKS.BRONZE_TSLA scheduled__2024-01-24T00:00:00+00:00 [running]> on host kaiwang.
[2024-09-15 23:28:31,976] {taskinstance.py:1569} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=BRONZER_GET_STONCKS
AIRFLOW_CTX_TASK_ID=BRONZE_TSLA
AIRFLOW_CTX_EXECUTION_DATE=2024-01-24T00:00:00+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2024-01-24T00:00:00+00:00
[2024-09-15 23:28:35,919] {history.py:244} ERROR - $TSLA: possibly delisted; no price data found  (1h 2024-01-21 -> 2024-01-03) (Yahoo error = "Invalid input - start date cannot be after end date. startDate = 1705813200, endDate = 1704258000")
[2024-09-15 23:28:35,921] {python.py:173} INFO - Done. Returned value was: None
[2024-09-15 23:28:35,925] {taskinstance.py:1395} INFO - Marking task as SUCCESS. dag_id=BRONZER_GET_STONCKS, task_id=BRONZE_TSLA, execution_date=20240124T000000, start_date=20240916T022831, end_date=20240916T022835
[2024-09-15 23:28:35,948] {local_task_job.py:156} INFO - Task exited with return code 0
[2024-09-15 23:28:35,957] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
